win32/src/job_object_helper.c

```diff
@@ -11,7 +11,130 @@
 
 IMPLEMENT_MOCKABLE_FUNCTION(, int, job_object_helper_limit_resources, uint32_t, percent_physical_memory, uint32_t, percent_cpu)
 {
-    (void) percent_physical_memory;
-    (void) percent_cpu;
-    return MU_FAILURE;
+    int result;
+    LogInfo("job_object_helper_limit_resources called with parameters:"
+        " uint32_t percent_physical_memory = %d"
+        ", uint32_t percent_cpu = %d",
+        percent_physical_memory,
+        percent_cpu);
+
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_001: [ If percent_physical_memory is 0, job_object_helper_limit_resources shall fail and return a non-zero value. ] */
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_002: [ If percent_cpu is 0, job_object_helper_limit_resources shall fail and return a non-zero value. ] */
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_004: [ If percent_physical_memory is greater than 100, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+    /*Codes_SRS_JOB_OBJECT_HELPER_18_005: [ If percent_cpu is greater than 100, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+    if ((percent_physical_memory == 0) ||
+        (percent_physical_memory > 100) || 
+        (percent_cpu == 0) ||
+        (percent_cpu > 100))
+    {
+        LogError("Invalid arguments:"
+            " uint32_t percent_physical_memory = %d"
+            ", uint32_t percent_cpu = %d",
+            percent_physical_memory,
+            percent_cpu);
+        /*Codes_SRS_JOB_OBJECT_HELPER_18_015: [ If there are any failures, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+        result = MU_FAILURE;
+    }
+    else
+    {
+        /*Codes_SRS_JOB_OBJECT_HELPER_18_006: [ job_object_helper_limit_resources shall call CreateJobObject to create a new job object passing NULL for both lpJobAttributes and lpName. ]*/
+        HANDLE job_object = CreateJobObject(NULL, NULL);
```

Comment:
>job_object [](http://example.com/codeflow?start=15&length=10)

This job_object should stay behind a HANDLE of the module

Reply:
next time.